### PR TITLE
v0.4.0 release: visualization module and orbital parameter search

### PR DESCRIPTION
**Pull request by JessicaS11**
Lots of great work included here that we want to make available via conda and pip:
- improved query object area of interest visualization (if you install geoviews)
- visualization module, including an interactive quick-view of elevation data using the OpenAltimetry API
- ability to query data by orbital parameters
- improved NSIDC message surfacing
- transition links to match transition from master to main
- improve formatting, reduce warnings

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -53,11 +59,11 @@ Examples (Jupyter Notebooks)
...
 Listed below are example jupyter-notebooks
 
-`ICESat-2_DAAC_DataAccess_Example <https://github.com/icesat2py/icepyx/blob/master/examples/ICESat-2_DAAC_DataAccess_Example.ipynb>`_
+`ICESat-2_DAAC_DataAccess_Example <https://github.com/icesat2py/icepyx/blob/main/examples/ICESat-2_DAAC_DataAccess_Example.ipynb>`_
 
-`ICESat-2_DAAC_DataAccess2_Subsetting <https://github.com/icesat2py/icepyx/blob/master/examples/ICESat-2_DAAC_DataAccess2_Subsetting.ipynb>`_
+`ICESat-2_DAAC_DataAccess2_Subsetting <https://github.com/icesat2py/icepyx/blob/main/examples/ICESat-2_DAAC_DataAccess2_Subsetting.ipynb>`_
 
-`ICESat-2_DEM_comparison_Colombia_working <https://github.com/icesat2py/icepyx/blob/master/examples/ICESat-2_DEM_comparison_Colombia_working.ipynb>`_
+`ICESat-2_DEM_comparison_Colombia_working <https://github.com/icesat2py/icepyx/blob/main/examples/ICESat-2_DEM_comparison_Colombia_working.ipynb>`_
```
Comment (weiji14) [inline README.rst:68]: Add a link on the main README.md to Tian's Data Visualization example here.
```suggestion
`ICESat-2 Data Visualization Example <https://github.com/icesat2py/icepyx/blob/main/examples/ICESat-2_Data_Visualization_Example.ipynb>`_

```

**Review by weiji14 — APPROVED** [review]
Just one suggestion, otherwise let's not delay this release any further!

File: doc/source/user_guide/changelog/v0.4.0.rst [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+- New example to showcase new visualization module
+
+
+Other
+^^^^^
+
+- Add geoviews as extra optional dependency in setup.py (#193)
+- Bulk rename master to main (#194)
+- surface NSIDC order and http download errors more effectively to user (#195)
+
```
Comment (JessicaS11) [inline doc/source/user_guide/changelog/v0.4.0.rst:44]:
```suggestion
- switch to using a miniconda environment install for travis testing
```

**Comment by weiji14** [comment]
FYI, the conda-forge autobot has opened up https://github.com/conda-forge/icepyx-feedstock/pull/2 for icepyx v0.4.0, I've updated the dependencies (and included `geoviews` so people get interactivity out of the box) in the conda-forge recipe and will merge it once the tests pass.

**Comment by JessicaS11** [comment]
Thanks for taking care of this @weiji14!

**Pull request merged**